src/ci/docker/run.sh

```diff
@@ -199,6 +199,10 @@ else
     args="$args --env SCCACHE_DIR=/sccache --volume $HOME/.cache/sccache:/sccache"
 fi
 
+if [ "$READ_ONLY_SRC" != "0" ]; then
```

Comment:
This is a bit "dangerous", in that if you actually overwrite something from inside the Docker image (which runs as root by default), it might break builds outside of the container. I used `--dev` to create a shell inside of an image (I had no idea about `--dev` btw, great thing! We should document these things somewhere) and ran `x.py check`. Then I exited the container and run it against (outside of the container), and got this error:
```bash
thread 'main' panicked at src/core/build_steps/compile.rs:1792:5:
fs::create_dir_all(&self_contained_dst) failed with Permission denied (os error 13)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I had to run `sudo rm -rf build` to fix things.

Reply:
This only works when we explicitly set that env to 0. If your container runs under the root mode, yeah, you shouldn't do that. But for people who runs containers without the root mode, this problem will not happen. By default, volumes are still read-only and when people want to be able to build tools in containers I think they should be able to turn off that read-only limitation.

Reply:
Yeah, I know that it's not enabled by default. I guess that setting this env is "expert mode" which enables advanced behavior (the possibility to overwrite local data from inside the image), and in that case it will be the responsibility of the user to fix this. Maybe we could at least include some comment on top of the ENV var to describe what is it for, what are the potential "dangers" (breaking the `build` directory), and how to fix it (`sudo rm -rf build`).

Reply:
Yeah sure, additionally I can print some warnings to stdout. What do you think?

Reply:
I think that a comment is enough. Because probably no one will use this env var "by accident" (in that case it would be useful to see some warning in stdout), since users need to go to this file and actually find this env var to even realize that they can turn off the read only mode.